plugins/user_mapping/src/user_claim.cpp

```diff
@@ -25,40 +26,48 @@ namespace
 
 	auto match(const nlohmann::json& _params) -> std::optional<std::string> {
 		if (auto claim{_params.find(claim_to_match)}; claim != std::end(_params)) {
-			return {*claim};
+			return claim->get<std::string>();
 		}
 
 		return std::nullopt;
 	} // match
 } // anonymous namespace
 
-auto user_mapper_init(const char* _args) -> void {
+auto user_mapper_init(const char* _args) -> int {
 	try {
 		spdlog::debug("{}: recieved _args [{}]", __func__, _args);
 		init(nlohmann::json::parse(_args));
 	} catch (const std::exception& e) {
 		spdlog::debug("{}: {}", __func__, e.what());
+		return 1;
 	}
+	return 0;
 } // user_mapper_init
 
-auto user_mapper_match(const char* _param) -> char* {
+auto user_mapper_match(const char* _param, char** _match) -> int {
 	spdlog::debug("{}: Attempting match of _param [{}]", __func__, _param);
 	try {
 		auto res{match(nlohmann::json::parse(_param))};
 
 		if (res) {
 			spdlog::debug("{}: matched _param [{}] to user [{}]", __func__, _param, *res);
-			char* matched_username{static_cast<char*>(std::malloc(res->size()))};
-			std::strncpy(matched_username, res->c_str(), res->size() + 1);
-			return matched_username;
+			char* matched_username{strdup(res->c_str())};
+
+			*_match = matched_username;
```

Comment:
May want to check for `nullptr` before dereferencing.

Reply:
Will probably end up doing for all interface functions & args.

Reply:
Done.